services/packagist/packagist-dependency-version.service.js

```diff
@@ -0,0 +1,204 @@
+    pattern: ':user/:repo/:dependency+',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Packagist Dependency Version Support',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (specify version)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        version: 'v2.8.0',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '1.12',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+    {
+      title: 'Packagist PHP Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'php',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+        version: 'v2.8.0',
+      },
+      staticPreview: this.render({
+        dependency: 'php',
+        dependencyVersion: '^7.1.3',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'dependency version',
+    color: 'blue',
+  }
+
+  static render({ dependency, dependencyVersion }) {
+    return {
+      label: dependency,
+      message: dependencyVersion,
+    }
+  }
+
+  async getDependencyVersion({
+    json,
+    user,
+    repo,
+    dependency,
+    version = '',
+    server,
+  }) {
+    let packageVersion
+    const versions = BasePackagistService.expandPackageVersions(
+      json,
+      this.getPackageName(user, repo)
+    )
+
+    if (version === '') {
+      packageVersion = this.findLatestRelease(versions)
+    } else {
+      try {
+        packageVersion = await this.findSpecifiedVersion(
+          versions,
+          user,
+          repo,
+          version,
+          server
+        )
+      } catch (e) {
+        packageVersion = null
+      }
+    }
+
+    if (!dependency) {
+      throw new NotFound({
+        prettyMessage: 'dependency vendor or repo not specified',
+      })
+    }
+
+    if (!packageVersion) {
+      throw new NotFound({ prettyMessage: 'invalid version' })
+    }
+
+    if (!packageVersion.require) {
+      throw new NotFound({ prettyMessage: 'version requirement not found' })
+    }
+
+    const depLowerCase = dependency.toLowerCase()
```

Comment:
Are we sure no packagist vendor name, package name or extension name can contain an upper-case character?

Reply:
Good question - I don’t think we can be sure of that.
Should i remove the above line?
It would then have issues with querries containing capital letters 
(https://github.com/badges/shields/pull/8371#issuecomment-1241246084)

We can do one of the following:
1) first check if there is a position in the response in ‚require’ section that contains directly the dependecy’s name (with capital letters)
and if not, transform the queried dependency’s name to lowercase and repeat the comparison
2) transform every dependency’s name from the reeponse to lowercase - and compare with lowercase depemdency’s name from the query.

What would be your preference?

Reply:
I modified it, now it looks like this - is it fine?

```js

    // All dependencies' names in the 'require' section from the response should be lowercase,
    // so that we can compare lowercase name of the dependency given via url by the user.
    Object.keys(packageVersion.require).forEach(dependency => {
      packageVersion.require[dependency.toLowerCase()] =
        packageVersion.require[dependency]
    })

    const depLowerCase = dependency.toLowerCase()

    if (!packageVersion.require[depLowerCase]) {
      throw new NotFound({ prettyMessage: 'version requirement not found' })
    }

    return { dependencyVersion: packageVersion.require[depLowerCase] }
```